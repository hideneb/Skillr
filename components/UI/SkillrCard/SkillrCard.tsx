import React from 'react';
import Link from 'next/link';
import { SkillrDto } from '../../../lib/types/skillr';
import { authedFetch } from '../../../lib/authed-fetch';

type SkillrCardProps = {
    skillr: SkillrDto;
    isLoggedIn: boolean;
};

const SkillrCard: React.FC<SkillrCardProps> = ({ skillr, isLoggedIn }) => {
    const [isFavorite, setIsFavorite] = React.useState(skillr.isFavourite || false);

    const addFavorite = (skillrId: string) => {
        return authedFetch(`/api/favorites/${skillrId}`, {
            method: 'POST',
        })
            .then((res) => res.json())
            .then(() => setIsFavorite(true));
    };
    const removeFavorite = (skillrId: string) => {
        return authedFetch(`/api/favorites/${skillrId}`, {
            method: 'DELETE',
        })
            .then((res) => res.json())
            .then(() => setIsFavorite(false));
    };

    return (
        <span key={skillr.id}>
            {isLoggedIn ? (
                isFavorite ? (
                    <span className="heart-full" onClick={() => removeFavorite(skillr.id)} />
                ) : (
                    <span className="heart-empty" onClick={() => addFavorite(skillr.id)} />
                )
            ) : (
                ''
            )}
            {skillr.isAvailableNow && (
                <span
                    style={{
                        backgroundColor: 'green',
                        borderRadius: '50%',
                        height: '10px',
                        width: '10px',
                        display: 'inline-block',
                        marginRight: '5px',
                    }}
                />
            )}
            <img
                src={skillr.profileImage}
                style={{ height: '150px', width: '200px' }}
                alt={`Profile image for @${skillr.username}`}
            />
            <Link href="/skillrs/[id]" as={`/skillrs/${skillr.id}`}>
                <a>@{skillr.username}</a>
            </Link>
            <sub>{skillr.skills[0].skill.name}</sub>
        </span>
    );
};

export default SkillrCard;
