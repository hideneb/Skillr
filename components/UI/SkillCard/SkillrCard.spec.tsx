import React from 'react';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';

import '@testing-library/jest-dom';
import SkillrCard from '../SkillrCard';

test('displaying a skillr', async () => {
    // Arrange

    const { container } = render(
        <SkillrCard
            skillr={{
                id: '1',
                username: 'username',
                displayName: '',
                profileImage: 'https://via.placeholder.com/150',
                about: '',
                instagram: '',
                linkedin: '',
                isAvailableNow: true,
                skills: [
                    {
                        skillrSkillId: '1',
                        skillId: 1,
                        skill: {
                            id: 1,
                            lightIcon: '',
                            lightIconFilename: '',
                            darkIcon: '',
                            darkIconFilename: '',
                            popularIcon: '',
                            popularIconFilename: '',
                            name: 'Skillr skill name',
                            description: '',
                            disabled: false,
                            createdAt: new Date().toISOString(),
                            updatedAt: new Date().toISOString(),
                            parentId: -1,
                        },
                        ratePerMinute: 10,
                        createdAt: new Date().toISOString(),
                        submitDate: new Date().toISOString(),
                        preApprovedRate: true,
                        state: 3,
                        brief: '',
                        credentials: [],
                        isInstantMatch: false,
                        showcases: [],
                        showcaseLinks: [],
                    },
                ],
                rating: 0,
                localAvailability: undefined,
            }}
            isLoggedIn={false}
        />
    );

    // Act

    // Assert

    expect(await screen.findByText(/\@username/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});
