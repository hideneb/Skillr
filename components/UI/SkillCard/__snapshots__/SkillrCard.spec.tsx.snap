// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displaying a skillr 1`] = `
<div>
  <span>
    <span
      style="background-color: green; border-radius: 50%; height: 10px; width: 10px; display: inline-block; margin-right: 5px;"
    />
    <img
      alt="Profile image for @username"
      src="https://via.placeholder.com/150"
      style="height: 150px; width: 200px;"
    />
    <a
      href="/skillrs/1"
    >
      @
      username
    </a>
    <sub>
      Skillr skill name
    </sub>
  </span>
</div>
`;
